app/models/project.rb

```diff
@@ -12,10 +12,19 @@ def approve
     update(status: "approved")
   end
 
+  def reject
+    return true if rejected?
+    update(status: "rejected")
+  end
+
   def approved?
     status == "approved"
   end
 
+  def rejected?
+    status == "rejected"
+  end
```

Comment:
I feel like I should be using acts_as_state_machine or something similar for this?

Reply:
Let's not use acts_as_state_machine, it's a one-way door to callback hell. This is a productive ugly and the correct pattern will emerge if we leave it be.